lib/GetStream/StreamChat/Client.php

```diff
@@ -817,13 +817,28 @@ public function search($filterConditions, $query, $options=null)
         if ($options === null) {
             $options = [];
         }
+
+        if (array_key_exists('offset', $options) && $options['offset'] > 0) {
+            if (array_key_exists('next', $options) || array_key_exists('sort', $options)) {
+                throw new StreamException("Cannot use offset with next or sort parameters");
+            }
+        }
+
         $options['filter_conditions'] = $filterConditions;
         if (is_string($query)) {
             $options['query'] = $query;
         } else {
             $options['message_filter_conditions'] = $query;
         }
 
+        $sortFields = [];
+        if (array_key_exists('sort', $options)) {
+            $sort = $options['sort'];
+            foreach ($sort as $k => $v) {
+                $sortFields[] = ["field" => $k, "direction" => $v];
+            }
+        }
+        $options['sort'] = $sortFields;
```

Comment:
not sure if needed, but might be better not to mutate given param

Reply:
tested it and it won't mutate the original. array assignment will copy an array and not modify the original, see the Example #15 Recursive and multi-dimensional arrays here: https://www.php.net/manual/en/language.types.array.php

Reply:
Tested too. Quite reverse of intuition but seems it always copies and reference is explicit.

```
<?php

function doSomething($options) {
    $sortFields = [];
    if (array_key_exists('sort', $options)) {
        $sort = $options['sort'];
        foreach ($sort as $k => $v) {
            $sortFields[] = ["field" => $k, "direction" => $v];
        }
    }
    $options['sort'] = $sortFields;
}


$options = ["sort" => [["created_at" => -1]]];


echo("before: ");
print_r($options);
doSomething($options);
echo("after: ");
print_r($options);
```